tracer/contrib/database/sqltraced/sqltraced.go

```diff
@@ -0,0 +1,367 @@
+// Package sqltraced provides a traced version of any driver implementing the database/sql/driver interface
+package sqltraced
+
+import (
+	"context"
+	"database/sql"
+	"database/sql/driver"
+	"fmt"
+	"reflect"
+	"strings"
+
+	log "github.com/cihub/seelog"
+
+	"github.com/DataDog/dd-trace-go/tracer"
+	"github.com/DataDog/dd-trace-go/tracer/ext"
+)
+
+// Register takes a driver and registers a traced version of this one.
+// However, user must take care not using the same name of the original driver
```

Comment:
rather than require that users choose a different name for their drivers (this can have subtle consequences elsewhere) can we just smartly recover from a panic caused by a bad call to `sql.Register`? in the case of calling `sqltraced.Register(...., "postgres")` after previously doing `sql.Register("postgres")`, we can just log something and return. leave the comment in there for users who care about working with both a 'traced' and 'untraced' driver in their codebase. but i think this will be a minority, so let's make the default behavior better